src/capi/object.cpp

```diff
@@ -146,7 +146,7 @@ extern "C" PyObject* PyObject_SelfIter(PyObject* obj) noexcept {
 extern "C" int PyObject_GenericSetAttr(PyObject* obj, PyObject* name, PyObject* value) noexcept {
     try {
         if (value == NULL)
-            delattrGeneric(obj, static_cast<BoxedString*>(name)->s, NULL);
```

Comment:
It'd be great to have functions start taking llvm::StringRef or Pyston::String rather than allocating a fresh std::string (doesn't have to be in this PR).